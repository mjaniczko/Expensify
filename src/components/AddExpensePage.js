import React from 'react'


const AddExpensePage = () => (
  <div>
    This is form my add expense component
  </div>
)


export default AddExpensePage