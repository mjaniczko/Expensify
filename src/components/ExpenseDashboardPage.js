import React from 'react'

const ExpenseDashboardPage = () => (
  <div>
    This is form my dashboard component
  </div>
)


export default ExpenseDashboardPage