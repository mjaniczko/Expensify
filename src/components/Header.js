import React from 'react'
import { NavLink } from 'react-router-dom'

const Header = () => (
  <header>
    <h1>Expensify</h1>
    <NavLink to='/' activeClassName="is-active" exact>HOME</NavLink>
    <NavLink to='/create' activeClassName="is-active">CREATE EXPENSE</NavLink>
    <NavLink to='/help' activeClassName="is-active">HELP</NavLink>
  </header>
)

export default Header