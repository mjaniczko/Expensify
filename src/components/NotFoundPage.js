import React from 'react'
import { Link } from 'react-router-dom'

const NotFoundPage = () => (
  <div>
    404! Page not found - <Link to='/'>GO HOME</Link>
  </div>
)


export default NotFoundPage