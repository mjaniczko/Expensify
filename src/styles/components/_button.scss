.button {
  border: none;
  color: white;
  display: inline-block;
  background: $blue;
  font-weight: 300;
  line-height: 1;
  font-size: $font-size-large;
  padding: $s-size;
  text-decoration: none;
}

.button--secondary {
  background: #888;
}


.button--link {
  background: none;
}