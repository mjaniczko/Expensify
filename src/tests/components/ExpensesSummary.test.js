import React from 'react'
import { shallow } from 'enzyme'

import { ExpensesSummary } from '../../components/ExpensesSummary'

test('Should correcly render ExpensesSummary with 1 expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235}/>)
  expect(wrapper).toMatchSnapshot()
})

test('Should correcly render ExpensesSummary with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={2232123665}/>)
  expect(wrapper).toMatchSnapshot()
})