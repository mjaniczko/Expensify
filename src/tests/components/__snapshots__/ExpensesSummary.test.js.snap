// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correcly render ExpensesSummary with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totaling 
      <span>
        $2.35
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`Should correcly render ExpensesSummary with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        23
      </span>
       
      expenses
       totaling 
      <span>
        $22,321,236.65
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
